lib/rspec/core.rb

```diff
@@ -9,6 +9,46 @@
 
 RSpec::Support.define_optimized_require_for_rspec(:core) { |f| require_relative f }
 
+module RSpec
+  module Core
+    # @private
+    class LookupSet
+      include Enumerable
```

Comment:
This class has a larger interface than I expected we'd need.  Is the Enumerable include necessary?  What about `+`?  Are there places that uses all the methods provided here?

IMO, a small interface is a more easily supported one so I'd like to keep this to the minimum we actually need.